File: sBuilding/Trunk/Client/sBuilding/sNonResidentComponent/Interface/INonResidentMemberMap.cs
using sNonResidentComponent.Model;
using sNonResidentComponent.Model.CustomObj;
using System.Collections.Generic;

namespace sNonResidentComponent.Interface {
    public interface INonResidentMemberMap {
        NonResidentMemberMap Insert(string session, NonResidentMemberMap nonResMemMap);
        NonResidentMemberMap Update(string session, NonResidentMemberMap nonResMemMap);
        int Delete(string session, long nonMemMapId);
        NonResidentMemberMap Get(string session, long nonMemMapId);
        List<NonResidentMemberMapCustom> GetListAllMemMap(string session, long nonOrgId);
    }
}

File: sBuilding/Trunk/Server/Java/NonResidentService/src/com/swt/nonresident/INonResidentMemberMap.java
package com.swt.nonresident;

import java.util.List;

import com.swt.nonresident.customObject.NonResidentMemberMapCustom;
import com.swt.nonresident.domain.NonResidentMemberMap;

/**
 * Service quan ly member cho khach vang lai
 * 
 * @author MINH's Macbook Pro
 *
 */
public interface INonResidentMemberMap {
	/**
	 * Them member
	 * 
	 * @param nonResMemMap
	 * @return
	 */
	public NonResidentMemberMap insert(NonResidentMemberMap nonResMemMap);
	
	/**
	 * Sua member
	 * 
	 * @param nonResMemMap
	 * @return
	 */
	public NonResidentMemberMap update(NonResidentMemberMap nonResMemMap);
	
	/**
	 * Xoa member
	 * 
	 * @param nonMemMapId
	 * @return
	 */
	public int delete(long nonMemMapId);
	
	/**
	 * Lay member
	 * 
	 * @param nonMemMapId
	 * @return
	 */
	public NonResidentMemberMap get(long nonMemMapId);
	
	/**
	 * Lay tat ca cac danh sach member cua khach vang lai theo nonOrgId
	 * 
	 * @param nonOrgId
	 * @return
	 */
	public List<NonResidentMemberMapCustom> getListAllMemMap(long nonOrgId);
}
